File: app/src/main/java/kr/djspi/pipe01/dto/PipeType.java
package kr.djspi.pipe01.dto;

import java.io.Serializable;

import kr.djspi.pipe01.R;
import lombok.Data;

@Data
public class PipeType implements DataItem, Serializable {

    public int id;
    public String header;
    public String pipe;
    public String unit;

    public static PipeTypeEnum parsePipeType(String name) {
        for (PipeTypeEnum pipe : PipeTypeEnum.values()) {
            if (pipe.name.equals(name)) return pipe;
        }
        return PipeTypeEnum.기타관로;
    }

    static final String HEAD_NULL = "";
    static final String HEAD_RAD = "관경";
    static final String HEAD_CORE = "코어수";
    static final String UNIT_MM = "mm";
    static final String UNIT_CORE = "코어";

    /**
     * Enum 목록의 순서는 변경하지 않는다.
     */
    @SuppressWarnings("NonAsciiCharacters")
    public enum PipeTypeEnum implements Serializable {
        도시가스("도시가스", R.drawable.cir_01_map, HEAD_RAD, UNIT_MM),
        상수관로("상수관로", R.drawable.cir_02_map, HEAD_RAD, UNIT_MM),
        하수관로("하수관로", R.drawable.cir_03_map, HEAD_RAD, UNIT_MM),
        오수관로("오수관로", R.drawable.cir_04_map, HEAD_RAD, UNIT_MM),
        전기관로("전기관로", R.drawable.cir_05_map, HEAD_CORE, UNIT_CORE),
        통신관로("통신관로", R.drawable.cir_06_map, HEAD_CORE, UNIT_CORE),
        난방관로("난방관로", R.drawable.cir_07_map, HEAD_RAD, UNIT_MM),
        유류관로("유류관로", R.drawable.cir_08_map, HEAD_RAD, UNIT_MM),
        가로등("가로등", R.drawable.cir_09_map, HEAD_CORE, UNIT_CORE),
        CCTV("CCTV", R.drawable.cir_10_map, HEAD_CORE, UNIT_CORE),
        광케이블("광케이블", R.drawable.cir_11_map, HEAD_CORE, UNIT_CORE),
        기타관로("기타관로", R.drawable.cir_12_map, HEAD_RAD, UNIT_MM);

        private String name;
        private int drawRes;
        private String header;
        private String unit;

        public String getName() {
            return name;
        }

        public int getDrawRes() {
            return drawRes;
        }

        public String getHeader() {
            return header;
        }

        public String getUnit() {
            return unit;
        }

        PipeTypeEnum(String name, int drawRes, String header, String unit) {
            this.name = name;
            this.drawRes = drawRes;
            this.header = header;
            this.unit = unit;
        }
    }
}

File: app/src/main/java/kr/djspi/pipe01/dto/PipeType.kt
package kr.djspi.pipe01.dto

import kr.djspi.pipe01.R
import java.io.Serializable

data class PipeType(var id: Int = -1) : DataItem, Serializable {

    var header: String? = null
    var pipe: String? = null
    var unit: String? = null

    /**
     * Enum 목록의 순서는 변경하지 않는다.
     */
    @Suppress("NonAsciiCharacters", "EnumEntryName")
    enum class PipeTypeEnum(
        internal val pipeName: String,
        internal val drawRes: Int,
        internal val drawResValve: Int,
        internal val header: String,
        internal val unit: String
    ) : Serializable {
        가스관로("가스관로", R.drawable.cir_01_map, R.drawable.cir_01_map_valve, "관경", "mm"),
        상수관로("상수관로", R.drawable.cir_02_map, R.drawable.cir_02_map_valve, "관경", "mm"),
        하수관로("하수관로", R.drawable.cir_03_map, R.drawable.cir_03_map_valve, "관경", "mm"),
        오수관로("오수관로", R.drawable.cir_04_map, R.drawable.cir_04_map_valve, "관경", "mm"),
        전기관로("전기관로", R.drawable.cir_05_map, R.drawable.cir_05_map_valve, "관경", ""),
        통신관로("통신관로", R.drawable.cir_06_map, R.drawable.cir_06_map_valve, "관로수", ""),
        난방관로("난방관로", R.drawable.cir_07_map, R.drawable.cir_07_map_valve, "관경", "mm"),
        유류관로("유류관로", R.drawable.cir_08_map, R.drawable.cir_08_map_valve, "관경", "mm"),
        가로등("가로등", R.drawable.cir_09_map, R.drawable.cir_09_map_valve, "코어수", "코어"),
        CCTV("CCTV", R.drawable.cir_10_map, R.drawable.cir_10_map_valve, "코어수", "코어"),
        광케이블("광케이블", R.drawable.cir_11_map, R.drawable.cir_11_map_valve, "관로수", ""),
        기타관로("기타관로", R.drawable.cir_12_map, R.drawable.cir_12_map_valve, "관경", "mm"),
        우수관로("우수관로", R.drawable.cir_12_map, R.drawable.cir_12_map_valve, "관경", "mm");

        companion object {
            fun parsePipeType(name: String): PipeTypeEnum {
                values().forEach {
                    if (it.pipeName == name) return it
                }
                return 기타관로
            }
        }
    }
}
